tests/framework/extensions/hardening/k3s/harden_nodes.go

```diff
@@ -0,0 +1,81 @@
+package hardening
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/sirupsen/logrus"
+)
+
+func CheckClusterHardening(client *rancher.Client, hardened bool, nodes []*nodes.Node, nodeRoles []string) error {
+	if hardened {
+		for key, node := range nodes {
+			logrus.Infof("Setting kernel parameters on node %s", node.NodeID)
+			_, err := node.ExecuteCommand("sudo setenforce 1")
+			if err != nil {
+				return err
+			}
+			_, err = node.ExecuteCommand("sudo bash -c 'echo vm.panic_on_oom=0 >> /etc/sysctl.conf'")
+			if err != nil {
+				return err
+			}
+			_, err = node.ExecuteCommand("sudo bash -c 'echo kernel.panic=10 >> /etc/sysctl.conf'")
+			if err != nil {
+				return err
+			}
+			_, err = node.ExecuteCommand("sudo bash -c 'echo kernel.panic_on_oops=1 >> /etc/sysctl.conf'")
+			if err != nil {
+				return err
+			}
+			_, err = node.ExecuteCommand("sudo bash -c 'echo kernel.keys.root_maxbytes=25000000 >> /etc/sysctl.conf'")
+			if err != nil {
+				return err
+			}
+			_, err = node.ExecuteCommand("sudo bash -c 'sysctl -p /etc/sysctl.conf'")
+			if err != nil {
+				return err
+			}
+
+			if nodeRoles[key] == "--etcd --controlplane --worker" || nodeRoles[key] == "--controlplane" {
```

Comment:
Why wouldn't this work for the dynamic input? Could I get a little clarification. Thank you!

Reply:
Oh, because the dynamic input uses `machinepools.NodeRoles`. The non-dynamic input defines the node doesn't, so one of two things would have to happen.

Either we convert the non-dynamic input to use `machinepools.NodeRoles` and I modify the file to work for both, or just run only the non-dynamic input. I chose the latter option just cause it was getting a bit hairy to do the former option.

I can definitely take another look at this if we think its better to have both for consistency sake

Reply:
https://github.com/rancher/rancher/blob/58627999b77516dc1e14e0294a14d9154fd60054/tests/v2/validation/provisioning/k3s/custom_cluster_test.go#L183-L196

Reply:
Let me take a double-take on that , it's probably the case I was doing too much at once 😄 . Cause yeah, that seems like it should take without issue looking at it further. Thanks man!

Reply:
Thank you!!